src/reducers/starWarsReducer.js

```diff
@@ -1,10 +1,33 @@
-import /* we need our action types here*/ "../actions";
+import { GET_SWAPI, ERROR, SUCCESS } from "../actions"; /* we need our action types here*/
 const initialState = {
-  characters: []
+    characters: [],
+    loading: false,
+    error: null
   // Array characters, Boolean fetching, null error.
 };
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
+    case GET_SWAPI:
+      return {
+        ...state,
+        loading: true,
+        characters: [],
```

Comment:
you dont need to declare the characters value to an empty array here in this method as you have already declared it in the initalState object